pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,266 @@
+
+    if 2**len(wires) != len(features):
+        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**n_qubits which is {}, "
+                         "got {}.".format(2 ** len(wires), len(features)))
+
+    QubitStateVector(features, wires=wires)
+
+
+def SqueezingEmbedding(features, wires, execution='amplitude', c=0.1):
+    r"""
+    Encodes the entries of ``features`` into the squeezing phases :math:`\phi` or amplitudes :math:`r` of the modes of
+    a continuous-variable quantum state.
+
+    The mathematical definition of the squeezing gate is given by the operator
+
+    ..math::
+
+        S(z) = \exp\left(\frac{r}{2}\left(e^{-i\phi}\a^2 -e^{i\phi}{\ad}^{2} \right) \right),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``n_wires`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+        wires (int): List of qumode indices for the qumodes used for the embedding
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode squeezing, while
+            ``'amplitude'`` uses the amplitude.
+        c (float): parameter setting the value of the phase of all squeezing gates if ``execution='amplitude'``, or the
+            amplitude of all squeezing gates if ``execution='phase'`` to a constant.
+    """
+
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if len(wires) != len(features):
+        raise ValueError("SqueezingEmbedding requires a feature vector of size n_wires which is {}"
+                         ", got {}.".format(len(wires), len(features)))
+
+    for f, w in zip(features, wires):
+        if execution == 'amplitude':
+            Squeezing(f, c, wires=w)
+        elif execution == 'phase':
+            Squeezing(c, f, wires=w)
+        else:
+            raise ValueError("Execution strategy {} not known. Has to be `phase` or `amplitude`.".format(execution))
+
+
+def DisplacementEmbedding(features, wires, execution='amplitude', c=0.1):
+    r"""
+    Encodes the entries of ``features`` into the displacement phases :math:`\phi` or amplitudes :math:`r` of the modes of
+    a continuous-variable quantum state.
+
+    The mathematical definition of the displacement gate is given by the operator
+
+    ..math::
+            D(\alpha) = \exp(r (e^{i\phi}\ad -e^{-i\phi}\a)),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``n_wires`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+        wires (int): List of qumode indices for the qumodes used for the embedding
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode squeezing, while
```

Comment:
phase of single-mode displacement? rather than squeezing?

Reply:
Nice find!